File: backend/services/fine_tune.py
import openai

def start_fine_tune(file_path):
    """
    Inicia el proceso de fine-tuning con un archivo JSONL.
    """
    response = openai.fine_tuning.jobs.create(training_file=file_path, model="gpt-4o")
    return response['id']


def get_fine_tune_status(fine_tune_id):
    """
    Devuelve el estado del proceso de fine-tuning.
    """
    response = openai.fine_tuning.jobs.retrieve(id=fine_tune_id)
    return response['status']

File: frontend/src/services/fineTune.js
import API from './api';

// 📌 Iniciar Fine-Tuning
export const startFineTune = async (file) => {
    const formData = new FormData();
    formData.append("file", file);

    const response = await API.post("/fine-tune/start", formData, {
        headers: {
            "Content-Type": "multipart/form-data",
        },
    });

    return response.data;
};

// 📌 Verificar Estado
export const getFineTuneStatus = async (fineTuneId) => {
    const response = await API.get(`/fine-tune/status/${fineTuneId}`);
    return response.data;
};